src/modules/esl-anchornav/core/esl-anchornav.ts

```diff
@@ -0,0 +1,223 @@
+
+import type {DelegatedEvent, ESLIntersectionEvent} from '../../esl-event-listener/core';
+import type {ESLAnchorData, ESLAnchornavRender} from './esl-anchornav-types';
+
+/**
+ * ESLSAnchornav
+ * @author Dmytro Shovchko
+ *
+ * ESLAnchornav is a component that collects content anchors from the page and provides anchor navigation
+ */
+@ExportNs('Anchornav')
+export class ESLAnchornav extends ESLBaseElement {
+  public static override is = 'esl-anchornav';
+  public static _renderers: Map<string, ESLAnchornavRender> = new Map();
+
+  /** Default renderer for anchornav item */
+  public static defaultRenderer: ESLAnchornavRender = (data: ESLAnchorData) => `<a class="esl-anchornav-item" href="#${data.id}">${data.title}</a>`;
+
+  /** Gets renderer by name */
+  public static getRenderer(name: string): ESLAnchornavRender {
+    return this._renderers.get(name) || this.defaultRenderer;
+  }
+
+  /** Sets renderer */
+  public static setRenderer(renderer: ESLAnchornavRender): void;
+  public static setRenderer(name: string, renderer: ESLAnchornavRender): void;
+  public static setRenderer(name: string | ESLAnchornavRender, renderer?: ESLAnchornavRender): void {
+    if (typeof name !== 'string') return this.setRenderer('default', name);
+    if (typeof name === 'string' && renderer) this._renderers.set(name, renderer);
+  }
+
+  @prop('esl:anchornav:request') public REQUEST_EVENT: string;
+  @prop('esl:anchornav:activechanged') public ACTIVECHANGED_EVENT: string;
+  @prop('esl:anchornav:update') public UPDATE_EVENT: string;
+  @prop('[esl-anchor]') protected ANCHOR_SELECTOR: string;
+
+  /** Item renderer which is used to build inner markup */
+  @attr({defaultValue: 'default'}) public renderer: string;
+
+  protected _active: ESLAnchorData;
+  protected _anchors: ESLAnchorData[] = [];
+  protected _offset: number;
+
+  /** Active anchor */
+  public get active(): ESLAnchorData {
+    return this._active;
+  }
+  public set active(value: ESLAnchorData) {
+    if (this._active === value) return;
+    this._active = value;
+    this._onActiveChange(value);
+  }
+
+  /** Anchors list */
+  protected get $anchors(): HTMLElement[] {
+    return this._anchors.map(({$anchor}) => $anchor);
+  }
+
+  /** Anchornav offset */
+  public get offset(): number {
+    return this._offset || 0;
+  }
+  public set offset(value: number) {
+    if (this._offset === value) return;
+    this._offset = value;
+    memoize.clear(this, '$viewport');
+    this.$$on(this._onAnchorIntersection);
+  }
+
+  /** Anchornav item renderer */
+  protected get itemRenderer(): ESLAnchornavRender {
+    return ESLAnchornav.getRenderer(this.renderer);
+  }
+
+  /** Anchornav item selector */
+  protected get itemSelector(): string {
+    return `.${this.baseTagName}-item`;
+  }
+
+  /** Anchornav items */
+  protected get $items(): HTMLAnchorElement[] {
+    return [...this.querySelectorAll<HTMLAnchorElement>(this.itemSelector)];
+  }
```

Comment:
We should not relay on class selector to track inner items.
Ideally we should have an array of exact items.
E.g.
```
get $items() {
   return [...this.$itemsArea.children];
}
```

In case we need some marker we should use an attribute that we manually add to inner items.

Reply:
removed